### ISSUE TITLE
TimedMuteGag Plugin Question

### ISSUE THREAD

**Issue by lazhoroni** [hlstriker/SMPlugins#77]
I have installed the TimedMuteGag plugin, it did not work. I need to install another plugin for it to work.

**Comment by RussianRonin** [comment]
Correct. There are dependencies to a lot of the plugins which pretty much lead up to the DB Core. Everything here is integrated with one another. So you either need to get the plugins all setup or strip out the code you want to use and implement it into your own solution.

**Comment by lazhoroni** [comment]
Is the API_DatabaseCore.sp sufficient for this plugin to work? I compiled the API_DatabaseCore.sp file. If there are other files, which files?

**Comment by HymnsForDisco** [comment]
Its a bit of a pain to work through all the files, but check the #includes in the plugin you are trying to use.
The timed gag plugin requires:
```
#include "../../Libraries/TimedPunishments/timed_punishments"
#include "../../Libraries/Admins/admins"

```
If a plugin calls on any *natives* from another plugin, (the natives are declared in .inc files timed_punishments.inc), then that plugin must be running on the server at the time for the native call to go through.
so the plugin calls natives declared in both timed_punishments.inc, and admins.inc, and those natives are implemented in the plugin sources of API_Admins.sp and API_TimedPunishments.sp

If you then check the #includes of those plugins, you'll see that API_TimedPunishments also requires
```
#include "../DatabaseCore/database_core"
#include "../DatabaseUsers/database_users"
#include "../DatabaseUserSessions/database_user_sessions"
#include "../DatabaseMapSessions/database_map_sessions"
#include "../DemoSessions/demo_sessions"
#include "../WebPageViewer/web_page_viewer"
```

So you will need to have all those plugins running to get the full functionality of the API_TimedPunishments plugin.  You could test it without some of them, for example, without the webviewer api plugin, any calls to show a page will simply do nothing since the natives won't be bound to anything. (valve has removed the web viewer functionality from the game anyways)
For additional libraries like the DatabaseUserSessions, that will handle storing records of when the punishments were applied, which are used for auditing purposes on our community. At a minimum, you should be running the DatabaseCore and DatabaseUsers to ensure that the timed punishments are actually being stored in a table for the user.  You could simply add those other libraries one by one until the plugin is functioning to your satisfaction, though having some missing may cause you to have runtime errors here and there.

It is a bit of a burden to try and set up a single plugin like this since the plugins were all built as a conjoined platform for the swoobles community, so if you simply want a plugin that creates a timed mute/gag, you could look into the plugin for guidance and write your own

**Comment by lazhoroni** [comment]
I'm going to test shared files. Thank you for everything.

**Comment by lazhoroni** [comment]
https://github.com/lazhoroni/Mute-Gag 
I've added the plugin by editing it. It does not work even though I install the required add-ons. How do I run the TimedMuteGag.smx plug-in?

**Comment by HymnsForDisco** [comment]
Once sourcemod is installed to your server, the TimedMuteGag.smx plugin file must be placed in the `csgo\addons\sourcemod\plugins` folder.  The plugins in our repository must be compiled with an older veersion of the sourcemod compiler, which is included in the repository in the `_SourceMod 1.6 Compiler` folder

You can confirm which plugins are running on the server by typing the server command `sm plugins list`
